solutions/cf-1365D.mdx

```diff
@@ -134,5 +134,120 @@ int32_t main() {
 ```
 
 </CPPSection>
+	
+<JavaSection>
+	
+```java	
+import java.util.*;
+import java.io.*;
+
+public class SolveTheMaze {
+	static int[] rowMovement = {0, 1, 0, -1};  // right, down, left, up
+	static int[] columnMovement = {1, 0, -1, 0};
+	static char[][] grid;
+	static boolean[][] visited;
+	static int numRows;
+	static int numColumns;
+
+	static boolean inBoundaries (int newRow, int newColumn) {
+		if (newRow >= 0 && newRow < numRows) {
+			if (newColumn >= 0 && newColumn < numColumns) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	static void floodfill (int row, int column) {
+		visited[row][column] = true;
+
+		for (int x = 0; x < 4; x++) {  // the 4 directions
+			int newRow = row + rowMovement[x];
+			int newColumn = column + columnMovement[x];
+			if (inBoundaries(newRow, newColumn)) {  // in boundaries
+				// not visited, and not #
+				if (! visited[newRow][newColumn]) {
+					if (grid[newRow][newColumn] != '#') {
+						floodfill(newRow, newColumn);
+					}
+				}
+			}
+		}
+	}
+
+	static String solve () {
+		boolean ok = true;
+		// surround the bad
+		for (int i = 0; i < numRows; i++) {
+			for (int j = 0; j < numColumns; j++) {
+				if (grid[i][j] == 'B') {
+					for (int x = 0; x < 4; x++) {  // the 4 directions
+						int newRow = i + rowMovement[x];
+						int newColumn = j + columnMovement[x];
+						// check if it is in the boundaries
+						if (inBoundaries(newRow, newColumn)) {
+							if (grid[newRow][newColumn] == 'G') {
+								ok = false;
+							}
+							if (grid[newRow][newColumn] == '.') {
+								// turn it into #
+								grid[newRow][newColumn] = '#';
+							}
+						}
+					}
+				}
+			}
+		}
+
+		if (ok == false) {
+			return "No";
+		}
+		ok = true;
```

Comment:
shouldn't ok already be true here? so you don't need to set it to true again

and i think you can just return instead of setting ok?